### PR TITLE
 Add icon D lang

### PR DESCRIPTION
**Pull request by zrfisaac**
d

**Issue:** #7255
**Similarweb rank:** [124.8K](https://dlang.org/)

### Checklist
  - [X] I updated the JSON data in `_data/simple-icons.json`
  - [X] I optimized the icon with SVGO or SVGOMG
  - [X] The SVG `viewbox` is `0 0 24 24`

### Description
D is a general-purpose programming language with static typing, systems-level access, and C-like syntax. With the D Programming Language, write fast, read fast, and run fast.

### CODE REVIEW

File: _data/simple-icons.json [hunk 1]
```diff
@@ -2374,6 +2374,11 @@
             "hex": "F7DF1E",
             "source": "https://github.com/cytoscape/cytoscape.js/blob/97c27700feefe2f7b79fca248763049e9a0b38c6/documentation/img/cytoscape-logo.svg"
         },
+        {
+            "title": "D",
+            "hex": "B03931",
+            "source": "https://github.com/dlang/dlang.org/blob/master/images/dlogo.svg"
```
Comment (sisp) [inline _data/simple-icons.json:2380]: Please use a permalink to the logo file as described in the [contribution guidelines](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md):

> In any case, the commit hash should be part of the URL. On GitHub, you can get the correct URL by pressing `y` on the GitHub page you want to link to. You can get help at the [getting permanent links to files page](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files).

```suggestion
            "source": "https://github.com/dlang/dlang.org/blob/6d0e2e5f6a8249031cfd010e389b99ff014cd320/images/dlogo.svg"
```

**Review by sisp — CHANGES_REQUESTED** [review]
A number of unrelated commits leaked into this PR which makes the commit history in this PR a bit polluted. Could you remove them? I guess you could rebase the branch to the latest state of `develop`.

The original D icon has a quality issue: The vertical line segment at the left of the viewbox isn't exactly vertical, there is a small x-axis offset at the bottom of that line segment. Any opinions on how to proceed here, @simple-icons/maintainers?

File: _data/simple-icons.json (hunk 1 above)
Comment (sachinraja) [reply _data/simple-icons.json:2380]: Let's retain quality issues. If we want to fix those, they should be upstreamed first.

**Comment by zrfisaac** [comment]
@sisp has been a long time with this issue, should we wait or close it ? Is there anything that you think that I could do ?

**Review by dirien — APPROVED** [review]
Hi @zrfisaac, it's a perfect fit! Thanks for the contribution

image

**Pull request merged**